TS_3/task2.ts

```diff
@@ -0,0 +1,72 @@
+
+// Реализуйте в классе следующие методы:
+
+
+
+type ObjectWithId = { id: number };
+
+class CusmomerStorage<T extends ObjectWithId> {       
+  // Создайте дженерик класс Storage<T>, где T должен быть ограничен объектом, имеющим КАК МИНИМУМ {id: number}.
+  // Задача класса - хранить объекты типа Т в приватном массиве
+  private items: T[];                           
+
+  constructor(initialItems?: T[]) {
+    //   - constructor должен принимать необзятельный массив объектов соответствующего типа.
+    this.items = initialItems ? initialItems : [];
+  }                                
+//     Для типизации используйте Utility Types
+  add(item: T): void;                                     //   - add, принимающий либо объект типа Т, либо объект типа Т без id. Метод должен быть реализовать с помощью ПЕРЕГРУЗКИ.
+  add(item: Omit<T, "id">): void;
+  add(item: T | Omit<T, "id">): void {
+    if ("id" in item) {  
+      this.items.push(item); //     Если массив пришел - объекты запушить в хранилище. Если на вход подан объект с айди - запушить его в хранилище
+    } else {
+      const newItem = { ...item, id: this.generateId() };
+      this.items.push(newItem as T);
+    }
+  }
+
+  update(updatedItem: ObjectWithId & T): void {            //   - update, принимающий объект с айди и любым набором остальных ключей из типа Т.
+    this.items = this.items.filter((item) =>
```

Comment:
Consider to refactor this method, did not get expected result with this implementation
`The filter() method creates a shallow copy  of a portion of a given array, filtered down to just the elements from the given array that pass the test implemented by the provided function.`

```js
1. find obj by id
2. update found obj
```